### PR TITLE
Auth Backend: Add JWT to CloudflareAccessResult

### PR DESCRIPTION
**Pull request by tylerd-canva**
## Hey, I just made a Pull Request!

This is a follow up to https://github.com/backstage/backstage/pull/12234. It's a small change so I'd just open a PR directly to get feedback.

The change adds the request JWT to the `CloudflareAccessResult` so that it can then be accessed by the `authHandler()` and `signIn` `resolver()` functions. My use case for this is so that I can make calls to enrich the User's `ProfileInfo` and related entities on behalf of the user using their own JWT.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by tylerd-canva** [comment]
cc @renlord as the author of the Cloudflare Access provider

**Review by freben — COMMENTED** [review]
Looks good to me. @Rugvip any security or practical concerns with having this cached?

File: plugins/auth-backend/src/providers/cloudflare-access/provider.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ export type CloudflareAccessResult = {
   claims: CloudflareAccessClaims;
   cfIdentity: CloudflareAccessIdentityProfile;
   expiresInSeconds?: number;
+  jwt: string;
```
Comment (Rugvip) [inline plugins/auth-backend/src/providers/cloudflare-access/provider.ts:156]: I'm thinking I'd prefer `token` here, or `accessToken`/`idToken` if that is what it is. JWT is a bit of an implementation detail.

File: plugins/auth-backend/src/providers/cloudflare-access/provider.ts [hunk 2]
```diff
@@ -277,6 +278,7 @@ export class CloudflareAccessAuthProvider implements AuthProviderRouteHandlers {
         claims,
         cfIdentity,
         expiresInSeconds: claims.exp - claims.iat,
+        jwt,
```
Comment (Rugvip) [inline plugins/auth-backend/src/providers/cloudflare-access/provider.ts:281]: This is cached for the full duration of the token, which can cause some races towards the end of the expiration. In this case I think perhaps the best solution might be to not cache the token at all, but add the token to the returned object. Both on line 284 and 266.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Makes sense, thanks! 👍 

A couple of nits, and just to double check, the profile fetching you want to do is different from the existing identity profile fetch right?

**Comment by tylerd-canva** [comment]
Thanks for the review! Fixing it up now.

> the profile fetching you want to do is different from the existing identity profile fetch right?

Yes, it's a separate internal service.

**Comment by renlord** [comment]
> cc @renlord as the author of the Cloudflare Access provider

LGTM 👍

File: plugins/auth-backend/src/providers/cloudflare-access/provider.test.ts [hunk 3]
```diff
@@ -138,12 +138,28 @@ describe('CloudflareAccessAuthProvider', () => {
...
     teamName: 'foobar',
     resolverContext: {} as AuthResolverContext,
-    authHandler: async ({ claims }) => ({
-      profile: {
-        email: claims.email,
-      },
-    }),
-    signInResolver: async () => {
+    authHandler: async result => {
+      expect(result).toEqual(
```
Comment (tylerd-canva) [inline plugins/auth-backend/src/providers/cloudflare-access/provider.test.ts:142]: It's a bit awkward to put expectations in these test fixtures, but I wasn't sure of a cleaner way to accomplish this type of verification. Open to any suggestions :)

**Review by freben — APPROVED** [review]
Thanks!

**Comment by suuus** [comment]
Thanks for your contribution @tylerd-canva 🙌 Please let me know if you’d like some help to contribute further.

**Pull request merged**